pkg/local/vmis.go

```diff
@@ -22,63 +22,44 @@ import (
 
 	glog.V(6).Infof("Creating/updating Local (Management Cluster) VMI for VerrazzanoBinding %s", binding.Name)
 
-	// Construct the set of expected VMIs
-	newVMIs := newVMIs(binding, verrazzanoURI, enableMonitoringStorage)
+	// Construct the expected VMI
+	newVmi := createInstance(binding, verrazzanoURI, enableMonitoringStorage)
 
 	// Create or update VMIs
-	var vmiNames = []string{}
-	for _, newVmi := range newVMIs {
-		vmiNames = append(vmiNames, newVmi.Name)
-		existingVmi, err := vmiLister.VerrazzanoMonitoringInstances(newVmi.Namespace).Get(newVmi.Name)
-		if existingVmi != nil {
-			newVmi.Spec.Grafana.Storage.PvcNames = existingVmi.Spec.Grafana.Storage.PvcNames
-			newVmi.Spec.Prometheus.Storage.PvcNames = existingVmi.Spec.Prometheus.Storage.PvcNames
-			newVmi.Spec.Elasticsearch.Storage.PvcNames = existingVmi.Spec.Elasticsearch.Storage.PvcNames
-			specDiffs := diff.CompareIgnoreTargetEmpties(existingVmi, newVmi)
-			if specDiffs != "" {
-				glog.V(6).Infof("VMI %s : Spec differences %s", newVmi.Name, specDiffs)
-				glog.V(4).Infof("Updating VMI %s", newVmi.Name)
-				newVmi.ResourceVersion = existingVmi.ResourceVersion
-				_, err = vmoClientSet.VerrazzanoV1().VerrazzanoMonitoringInstances(newVmi.Namespace).Update(context.TODO(), newVmi, metav1.UpdateOptions{})
-			}
-		} else {
-			glog.V(4).Infof("Creating VMI %s", newVmi.Name)
-			_, err = vmoClientSet.VerrazzanoV1().VerrazzanoMonitoringInstances(newVmi.Namespace).Create(context.TODO(), newVmi, metav1.CreateOptions{})
-		}
-		if err != nil {
-			return err
+	existingVmi, err := vmiLister.VerrazzanoMonitoringInstances(newVmi.Namespace).Get(newVmi.Name)
+	if existingVmi != nil {
+		newVmi.Spec.Grafana.Storage.PvcNames = existingVmi.Spec.Grafana.Storage.PvcNames
+		newVmi.Spec.Prometheus.Storage.PvcNames = existingVmi.Spec.Prometheus.Storage.PvcNames
+		newVmi.Spec.Elasticsearch.Storage.PvcNames = existingVmi.Spec.Elasticsearch.Storage.PvcNames
+		specDiffs := diff.CompareIgnoreTargetEmpties(existingVmi, newVmi)
+		if specDiffs != "" {
+			glog.V(6).Infof("VMI %s : Spec differences %s", newVmi.Name, specDiffs)
+			glog.V(4).Infof("Updating VMI %s", newVmi.Name)
+			newVmi.ResourceVersion = existingVmi.ResourceVersion
+			_, err = vmoClientSet.VerrazzanoV1().VerrazzanoMonitoringInstances(newVmi.Namespace).Update(context.TODO(), newVmi, metav1.UpdateOptions{})
 		}
+	} else {
+		glog.V(4).Infof("Creating VMI %s", newVmi.Name)
+		_, err = vmoClientSet.VerrazzanoV1().VerrazzanoMonitoringInstances(newVmi.Namespace).Create(context.TODO(), newVmi, metav1.CreateOptions{})
 	}
-
-	// Delete VMIs that shouldn't exist
-	selector := labels.SelectorFromSet(map[string]string{constants.VerrazzanoBinding: binding.Name})
-	existingVMIsList, err := vmiLister.VerrazzanoMonitoringInstances("").List(selector)
```

Comment:
I wonder why we had these before? Was there a leftover VMI in default "" namespace?